tests: test_running_base_functionality() failed on CI

[There](https://github.com/tarantool/tt/blob/88477a626916a012fe0e3494adc2a65127768153/test/integration/running/test_running_cluster.py#L113) is currently failed/flaky CI test. We need to fix it.

```
                with open(os.path.join(run_dir, inst, 'tarantool.pid')) as f:
>                   assert pidByInstanceName[inst] != f.readline()
E                   AssertionError: assert '40972' != '40972'
E                    +  where '40972' = <built-in method readline of _io.TextIOWrapper object at 0x7fe31c598860>()
E                    +    where <built-in method readline of _io.TextIOWrapper object at 0x7fe31c598860> = <_io.TextIOWrapper name='/tmp/pytest-of-runner/pytest-0/test_running_base_functionalit1/small_cluster_app/var/run/storage-master/tarantool.pid' mode='r' encoding='UTF-8'>.readline

test/integration/running/test_running_cluster.py:113: AssertionError
----------------------------- Captured stdout call -----------------------------
 INSTANCE                           STATUS   PID    MODE  CONFIG               BOX      UPSTREAM 
 small_cluster_app:storage-master   RUNNING  40954  RO    startup_in_progress  loading  --       
 small_cluster_app:storage-replica  RUNNING  40955  RO    startup_in_progress  loading  --       

   • The Instance small_cluster_app:storage-master (PID = 41009) has been terminated.
   • The Instance small_cluster_app:storage-replica (PID = 41010) has been terminated.

=========================== short test summary info ============================
FAILED test/integration/running/test_running_cluster.py::test_running_base_functionality - AssertionError: assert '40972' != '40972'
 +  where '40972' = <built-in method readline of _io.TextIOWrapper object at 0x7fe31c598860>()
 +    where <built-in method readline of _io.TextIOWrapper object at 0x7fe31c598860> = <_io.TextIOWrapper name='/tmp/pytest-of-runner/pytest-0/test_running_base_functionalit1/small_cluster_app/var/run/storage-master/tarantool.pid' mode='r' encoding='UTF-8'>.readline
===== 1 failed, 583 passed, 76 skipped, 44 deselected in 858.55s (0:14:18) =====
Error: running "python3 -m pytest -m not slow and not slow_ee and not notarantool test/integration" failed with exit code 1
Error: Process completed with exit code 1.
```

[logs_29212819413.zip](https://github.com/user-attachments/files/17290605/logs_29212819413.zip)

--- a/test/integration/running/test_running_cluster.py
+++ b/test/integration/running/test_running_cluster.py
@@ -7,7 +7,8 @@ import pytest
 import yaml
 
 from utils import (control_socket, extract_status, get_tarantool_version,
-                   run_command_and_get_output, run_path, wait_file)
+                   run_command_and_get_output, run_path, wait_file,
+                   wait_pid_disappear)
 
 tarantool_major_version, tarantool_minor_version = get_tarantool_version()
 
@@ -98,6 +99,12 @@ def test_running_base_functionality(tt_cmd, tmpdir_with_cfg):
             text=True
         )
         start_output = instance_process.stdout.read()
+
+        # Need to wait for all instances to stop before start.
+        for inst in instances:
+            # Wait when PID that was fetched on start disappears.
+            wait_pid_disappear(os.path.join(run_dir, inst, 'tarantool.pid'),
+                               pidByInstanceName.get(inst))
 
         for inst in instances:
             assert f"Starting an instance [{app_name}:{inst}]" in start_output

--- a/test/utils.py
+++ b/test/utils.py
@@ -573,3 +573,17 @@ def wait_for_lines_in_output(stdout, expected_lines: list):
                 break
 
     return output
+
+
+@retry(Exception, tries=40, delay=0.5)
+def wait_pid_disappear(file, target_pid):
+    found = True
+    while found:
+        if not os.path.exists(file):
+            found = False
+            break
+        with open(file, "r") as f:
+            if target_pid not in f.readline():
+                found = False
+                break
+    assert not found

